src/settings.js

```diff
@@ -821,6 +821,9 @@ var DEBUG_TAGS_SHOWING = [];
 // For internal use only
 var ORIGINAL_EXPORTED_FUNCTIONS = [];
 
+// Internal: tracks the list of function signatures that are proxied between threads.
+var PROXIED_FUNCTION_SIGNATURES = [];
```

Comment:
this only includes signatures for MAIN_THREAD_EM_ASM functions, right? Should library function signatures go in here too? Or how do they work today?

Reply:
The flow of generating the invokers is a bit different for library functions and EM_ASM functions. `PROXIED_FUNCTION_SIGNATURES` only contains the signatures of EM_ASM functions. These signatures are collected from fastcomp backend build output.

Regular JS functions know the signatures already in jsifier.js at JS library parsing time, via the `foo_sig: 'vii',` directives. Proxy invokers are generated in jsifier.js on line 343:

```javascript
if (!proxiedFunctionInvokers[invokerKey]) proxiedFunctionInvokers[invokerKey] = generateProxiedCallInvoker(sig, synchronousCall);
```

whereas EM_ASM signatures are a bit farther down on line 548:

```javascript
        // Generate worker->main thread proxy function invokers for MAIN_THREAD_EM_ASM() signatures.
        for (var i in PROXIED_FUNCTION_SIGNATURES) {
          var invokerKey = PROXIED_FUNCTION_SIGNATURES[i];
          if (!proxiedFunctionInvokers[invokerKey]) {
            var sig_sync = invokerKey.split('_');
            proxiedFunctionInvokers[invokerKey] = generateProxiedCallInvoker(sig_sync[0], sig_sync[1] === 'sync');
          }
```

The `proxiedFunctionInvokers` dictionary in that file tracks the "unification" of the two, ensuring that only one invoker is generated for each signature, even if they overlap.

Reply:
Adjusted the comment for that to reflect that `PROXIED_FUNCTION_SIGNATURES` only contains EM_ASM function signatures.